inspirehep/ext/jinja_filters/record.py

```diff
@@ -355,6 +355,48 @@ def epoch_to_year_format(date):
 
     @app.template_filter()
     def construct_date_format(date):
-        starting_date = time.strftime('%Y-%m-%d', time.gmtime(int(date) / 1000.))
+        starting_date = time.strftime(
+            '%Y-%m-%d', time.gmtime(int(date) / 1000.))
         ending_date = time.strftime('%Y-12-31', time.gmtime(int(date) / 1000.))
         return starting_date + '->' + ending_date
+
+    @app.template_filter()
+    def author_urls(l, separator):
+        result = []
+        for name in l:
+            if 'name' in name:
+                url = '<a href="/search?ln=en&amp;cc=HepNames&amp;p=name:'\
+                      + name['name'] + '&amp;of=hd">' + name['name'] + '</a>'
+                result.append(url)
+        return separator.join(result)
+
+    @app.template_filter()
+    def ads_links(record):
+        import re
```

Comment:
So, yeah, it is a pattern we use in some cases. Usually standard library imports are fine at a the global level as they are not then imported every time the function is run. (say if it called several times). But I guess its fine in this case.